app/scripts/components/common/blocks/block-map.tsx

```diff
@@ -1,29 +1,42 @@
 import React, { useMemo, useState, useEffect } from 'react';
 import styled from 'styled-components';
-import { ProjectionOptions } from 'veda';
-
+import { DatasetDatumFnResolverBag, ProjectionOptions, datasets } from 'veda';
+import { MapboxOptions } from 'mapbox-gl';
+import * as dateFns from 'date-fns';
 import {
   convertProjectionToMapbox,
   projectionDefault,
   validateProjectionBlockProps
-} from '../mapbox/map-options/utils';
-import { BasemapId } from '../mapbox/map-options/basemaps';
-
+} from '../map/controls/map-options/projections';
+import { Basemap } from '../map/style-generators/basemap';
+import { LayerLegend, LayerLegendContainer } from '../mapbox/layer-legend';
+import MapCoordsControl from '../map/controls/coords';
+import MapMessage from '../mapbox/map-message';
+import { formatCompareDate, formatSingleDate, resolveConfigFunctions } from '../map/utils';
+import { BasemapId, DEFAULT_MAP_STYLE_URL } from '../map/controls/map-options/basemap';
 import { utcString2userTzDate } from '$utils/date';
-import MapboxMap, { MapboxMapProps } from '$components/common/mapbox';
+import Map, { Compare, MapControls } from '$components/common/map';
 import { validateRangeNum } from '$utils/utils';
 import { HintedError } from '$utils/hinted-error';
+import {
+  NavigationControl,
+  ScaleControl
+} from '$components/common/map/controls';
+import { Layer } from '$components/exploration/components/map/layer';
+import {
+  S_SUCCEEDED
+} from '$utils/status';
+import { TimelineDataset, TimelineDatasetSuccess } from '$components/exploration/types.d.ts';
+
+import { reconcileDatasets } from '$components/exploration/data-utils';
+import { datasetLayers } from '$components/exploration/data-utils';
+import { useReconcileWithStacMetadata } from '$components/exploration/hooks/use-stac-metadata-datasets';
```

Comment:
I wonder if importing the types, components and hooks from the Exploration will always work, especially if at some point these features diverge (`block-map` VS the usage of the map hooks in the `Exploration` feature which to me it looks like something different). We have (I think?) two remaining places to replace the old with the new map component and we might need some of the types, or reconciliation methods in there (this is just an assumption at this point). If that's the case, maybe it's better to have some common base atoms and hooks that update them, and then each feature could derive sub-atoms based on them if needed?

Reply:
a common directory would be great! especially since there are actually a few components using `/exploration` logic & hooks. We haven't gotten to that clean up yet but I think both @hanbyul-here and I are aware. I think we only care to replace the story block maps with the new component and the others i'm actually unsure about.

As part of the refactor we have a story to break out the Map core feature component ([here](https://github.com/NASA-IMPACT/veda-ui/issues/899)) so much of this cleanup can happen then but also I think it makes sense to chip away at this gradually due to scope of use.

What do you think though?